ui/components/multichain/permission-details-modal/permission-details-modal.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { flatten } from 'lodash';
+import { useSelector } from 'react-redux';
+import {
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Modal,
+  Box,
+  Text,
+  AvatarAccount,
+  ModalFooter,
+  Button,
+  IconName,
+  ButtonVariant,
+  AvatarAccountSize,
+  AvatarAccountVariant,
+  ModalBody,
+  ButtonSize,
+} from '../../component-library';
+import { getPermissionsForActiveTab, getUseBlockie } from '../../../selectors';
+import { shortenAddress } from '../../../helpers/utils/util';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getPermissionDescription } from '../../../helpers/utils/permission';
+import PermissionCell from '../../app/permission-cell';
+import { Identity } from '../connected-accounts-menu/connected-accounts-menu.types';
+import {
+  AlignItems,
+  Display,
+  JustifyContent,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+
+export const PermissionDetailsModal = ({
+  onClose,
+  onClick,
+  isOpen,
+  account,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  onClick: () => void;
+  account: Identity;
+}) => {
+  const t = useI18nContext();
+  const permissions = useSelector(getPermissionsForActiveTab);
+  const useBlockie = useSelector(getUseBlockie);
+  if (!permissions.length) {
```

Comment:
Should this check be in the `Menu` / `MenuItem` instead?  Otherwise we show the item but nothing renders.

Reply:
Yup! Updated here https://github.com/MetaMask/metamask-extension/pull/23605/commits/0ee53110c1f0550cccfad71c4fd506f85b3dd902